draft-ietf-taps-transport-security.md

```diff
@@ -844,6 +855,70 @@ and the per-message nonce in the clear. Everything else is encrypted.
 
 - An unreliable transport protocol such as UDP.
 
+## OpenVPN
+
+OpenVPN {{OpenVPN}} is a commonly used protocol designed to replace IPsec. A
+major goal of this protocol is to provide an easy-to-use VPN. OpenVPN
+encapsulates either IP packets or Ethernet frames within a secure tunnel and
+can run over UDP or TCP.
+
+### Protocol Description
+
+OpenVPN facilitates authentication using either a pre-shared static key or
+using X.509 certificates and TLS. In pre-shared key mode, OpenVPN derives
+keys for encryption and authentication directly from one or multiple static
+keys. In TLS mode, OpenVPN encapsulates a TLS handshake, in which both peers
+must present a certificate for authentication. After the handshake, both sides
+contribute random source material to derive keys for encryption and
+authentication using the TLS pseudo random function (PRF). OpenVPN provides the
+possibility to authenticate and encrypt the TLS handshake itself using a
+pre-shared key or passphrase. Furthermore, it supports rekeying using TLS.
+
+After authentication and key exchange, OpenVPN encrypts payload data, i.e., IP
+packets or Ethernet frames, and signs the payload using an HMAC function. The
+default cipher is BlowFish and the default message digest algorithm is SHA1,
+but an application can select an arbitrary cipher, key size, and message digest
+algorithm for the HMAC. OpenVPN peers may also support cipher negotiation
+(NCP). If both hosts allow, they can upgrade the cipher to AES-256-GCM.
+Blocks are padded to an even multiple of block size.
```

Comment:
Is this true even for AES-GCM?

Reply:
Not sure. I looked at the code for a bit now, and while the encrypt function does call a function named "cipher_ctx_final" which the header file describes as "Pads the final cipher block using PKCS padding", but I'm not sure if this is actually done for AES-GCM.
This is the header file for the "Data Channel Cryptography SSL library-specific backend interface", and the possible libraries are openssl and mbedtls... so I guess if these libraries pad the data, it is padded, and otherwise it isn't? 
As padding is not necessary for GCM, I'd assume that it isn't used, also https://wiki.openssl.org/index.php/EVP_Authenticated_Encryption_and_Decryption does not contain any reference to padding. Also confirmed here: http://openssl.6102.n7.nabble.com/AES-GCM-padding-td43598.html

So whether padding occurs or not depends on the used cipher...
Maybe we should just drop this sentence? ;)

Reply:
Yeah, I think so. It's an internal detail that perhaps is not relevant.